### ISSUE TITLE
App container healthcheck port

### ISSUE THREAD

**Issue by tumbl3w33d** [mattermost/mattermost-docker#395]
I noticed that my app container is shown by `docker ps` as unhealthy, because in the Dockerfile it says

`HEALTHCHECK CMD curl --fail http://localhost:8000 || exit 1`

and nothing listens on port 8000 if you configured it differently (only TLS on 8443 in my case). It would be desirable to make that port configurable via docker's `ARG` mechanism.

**Comment by jasimmonsv** [comment]
You have two options, chang ethe DockerFile HEALTHCHECK to whatever port you set it to, or change the Mattermost ListenAddress to 8000 through the config file, or ENV variable.
Configfile -> {"ServiceSettings": {"ListenAddress": 8000}}
or
ENV_VAR -> MM_SERVICESETTINGS_LISTENADDRESS=8000

**Comment by Obbi89** [comment]
I'd recommend updating the Dockerfile to reflect overrides somehow

https://github.com/mattermost/mattermost-docker/blob/668f3bfb04ce35c4a6d53a2cd8a648663680c89a/app/Dockerfile#L42 to
```Dockerfile

ENV MM_SERVICESETTINGS_SITEURL=http://localhost:8000
HEALTHCHECK CMD curl --fail $MM_SERVICESETTINGS_SITEURL || exit 1
```

works in my example but I think it's not the right way to go here. Maybe it would be an idea to add health check command to mattermost itself to allow it to be something like:

```Dockerfile
HEALTHCHECK CMD mattermost health-check || exit 1
```

**Comment by pichouk** [comment]
I agree that having a health check command directly in Mattermost could be interesting, but your workaround with the `ENV` directive seems ok to me.
